.husky/pre-commit

```diff
@@ -1,4 +1,6 @@
 #!/bin/sh
 . "$(dirname "$0")/_/husky.sh"
 
-yarn lint-staged
+# Specifying `config` manually avoids `lint-staged` scanning the entire repository, even `package.json` containing test suites.
+# Use `--no-stash` to skip backup files as we don't automatically fix lint errors.
+./node_modules/.bin/lint-staged --config package.json --no-stash
```

Comment:
```suggestion
yarn lint-staged --config package.json --no-stash
```

Reply:
```
PS F:\babel> Measure-Command {yarn lint-staged}  

Days              : 0
Hours             : 0
Minutes           : 0
Seconds           : 2
Milliseconds      : 773
Ticks             : 27731558
TotalDays         : 3.20967106481481E-05
TotalHours        : 0.000770321055555556
TotalMinutes      : 0.0462192633333333
TotalSeconds      : 2.7731558
TotalMilliseconds : 2773.1558


PS F:\babel> Measure-Command {./node_modules/.bin/lint-staged}

Days              : 0
Hours             : 0
Minutes           : 0
Seconds           : 0
Milliseconds      : 454
Ticks             : 4540090
TotalDays         : 5.2547337962963E-06
TotalHours        : 0.000126113611111111
TotalMinutes      : 0.00756681666666667
TotalSeconds      : 0.454009
TotalMilliseconds : 454.009
```
`yarn` adds an extra 2s, which I tend to avoid.

Reply:
Oh ok, that's fine. I'm surprised that it's this slow just to load a script.

Reply:
Yeah, this seems to be brought by yarn3, which creates a new virtual environment for all commands.

Reply:
FYI here is my benchmark results

```
time yarn lint-staged
→ No staged files found.
yarn lint-staged  0.62s user 0.10s system 114% cpu 0.634 total

time ./node_modules/.bin/lint-staged
→ No staged files found.
./node_modules/.bin/lint-staged  0.17s user 0.04s system 70% cpu 0.285 total
```

I am good either way if the startup time is less than 1 sec, but the Windows platform performance is a bit annoying. Maybe yarn PnP will save us some time here? /cc @merceyz

Reply:
Merge it first?

Reply:
@JLHwung If the startup time in this case is just Yarn it probably wont help, we are aware of the slowdowns though.
Ref https://github.com/yarnpkg/berry/issues/2575